packages/@dcos/sorting/index.ts

```diff
@@ -0,0 +1,65 @@
+export enum ComparisonResult {
+  BIGGER = 1,
+  EQUAL = 0,
+  SMALLER = -1
+}
+export type Comparator<T> = (a: T, b: T) => ComparisonResult;
+export type Getter<I, T> = (data: I) => T;
+type compareWithGetter<T> = <I>(getter: Getter<I, T>) => Comparator<I>;
+
+export function sort<T>(comparators: Array<Comparator<T>>, data: T[]): T[] {
+  return [...data].sort((a, b) => {
+    for (const comparator of comparators) {
+      const result = comparator(a, b);
+
+      if (result !== 0) {
+        return result;
+      }
+    }
+
+    return 0;
+  });
+}
+
+export function reverseComparator<T>(comparator: Comparator<T>): Comparator<T> {
+  return (a, b) => {
+    const previousResult = comparator(a, b);
+
+    if (previousResult === ComparisonResult.EQUAL) {
+      return previousResult;
+    } else {
+      return (-1 * previousResult) as ComparisonResult;
+    }
+  };
+}
+
+function compareStringValues(a: string, b: string): ComparisonResult {
+  // Does not always return -1 or 1: https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#Description
+  const result = a.localeCompare(b);
+  return result > 0
+    ? ComparisonResult.BIGGER
+    : result < 0
+      ? ComparisonResult.SMALLER
```

Comment:
I thought we were banning nested ternaries. https://github.com/dcos-labs/eslint-config/blob/master/index.js#L22 Please change